src/main/java/com/mongodb/jdbc/MongoSQLValue.java

```diff
@@ -0,0 +1,108 @@
+package com.mongodb.jdbc;
+
+import java.io.StringWriter;
+import java.sql.Date;
+import java.text.SimpleDateFormat;
+import org.bson.BsonValue;
+import org.bson.codecs.BsonValueCodec;
+import org.bson.codecs.EncoderContext;
+import org.bson.json.JsonMode;
+import org.bson.json.JsonWriterSettings;
+
+/**
+ * MongoSQLValue is a wrapper for BsonValue. The purpose of this class is to override the toString()
+ * method to produce the extended JSON representation of a BsonValue for types that correspond to
+ * JDBC Types.OTHER, rather than the java driver's default BsonValue.toString() output.
+ *
+ * <p>The driver's BsonValue class is abstract and intentionally cannot be extended by third
+ * parties. The driver explains this is to keep the BSON type system closed. Therefore, this class
+ * does not extend BsonValue, instead it contains a BsonValue member.
+ */
+public class MongoSQLValue {
+    static final JsonWriterSettings JSON_WRITER_SETTINGS =
+            JsonWriterSettings.builder().outputMode(JsonMode.RELAXED).build();
+    static final EncoderContext ENCODER_CONTEXT = EncoderContext.builder().build();
+
+    // dateFormat cannot be static due to a threading bug in the library.
+    protected SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+
+    private BsonValue v;
+
+    public MongoSQLValue(BsonValue v) {
+        this.v = v;
+    }
+
+    /** @return The underlying BsonValue */
+    public BsonValue getBsonValue() {
+        return this.v;
+    }
+
+    @Override
+    public String toString() {
+        if (this.v == null) {
+            return null;
+        }
+
+        switch (this.v.getBsonType()) {
+            case BOOLEAN:
+                return this.v.asBoolean().getValue() ? "true" : "false";
+            case DATE_TIME:
+                Date d = new Date(this.v.asDateTime().getValue());
+                return dateFormat.format(d);
+            case DECIMAL128:
```

Comment:
Same as for Date. This will not return the relaxed extended JSON representation.
The value we provide is : _121684.45864468_
The relaxed extended JSON representation is : _{"$numberDecimal": "121684.45864468"}_

Reply:
done